tachyon/zk/r1cs/constraint_system/quadratic_arithmetic_program.h

```diff
@@ -26,19 +26,15 @@ namespace tachyon::zk::r1cs {
 template <typename F>
 F EvaluateConstraint(const std::vector<Cell<F>>& cells,
                      absl::Span<const F> assignments) {
-  std::vector<F> sums = base::ParallelizeMap(
-      cells, [assignments](absl::Span<const Cell<F>> chunk) {
-        F sum;
-        for (const Cell<F>& cell : chunk) {
-          if (cell.coefficient.IsOne()) {
-            sum += assignments[cell.index];
-          } else {
-            sum += assignments[cell.index] * cell.coefficient;
-          }
-        }
-        return sum;
-      });
-  return std::accumulate(sums.begin(), sums.end(), F::Zero(), std::plus<>());
+  F sum;
+  for (const Cell<F>& cell : cells) {
+    if (cell.coefficient.IsOne()) {
+      sum += assignments[cell.index];
+    } else {
+      sum += assignments[cell.index] * cell.coefficient;
+    }
+  }
+  return sum;
```

Comment:
Could you make this into a separate commit?

Reply:
Actually, this is part of the commit, since this used parallel loop within in an outer parallel loop!